Clarify close-iteration output

pbrisbin opened the pull request:
I found the previous output hard to understand:

```
Completed
- new points: 19
- carried over points: 14
- new stories: 13
- carried over stories: 17
Incomplete
- points completed: 11
- carry over points: 22
- carry over stories: 14

44 / 66
```

Some of my confusions:

- What does "new" mean?
- Does "carried over" mean into our out of the closing iteration?
- Ditto "carry over"
- How does "Incomplete" have "completed" points?
- is "44 / 66" velocity? Why make me do the math?

The new output is this:

```
Completed points: 33 (14 were carry-over)
Completed tasks: 30 (17 were carry-over)

Incomplete points: 22
Incomplete tasks: 14 (33 total cost)

Velocity: 44/66 (66.66666666666666%)
```

This attempts to clarify:

- That we're showing totals and breaking out what was carried
- Past tense when talking about carried into the iteration
- How we get credit for partial points, but show total incomplete too

And computes velocity for you.

Also included:

f99bd76 Ensure trailing newline in output

The fold was placing a leading newline, desired to separate the totals from the
logging above. It did not include a trailing newline, which messes up my prompt
when the tool is run.

We could explicitly do that, but it was easier to switch the fold to place
trailing newlines (including a final one) and then have an explicitly empty
first element in our list of lines.

pbrisbin committed:
Ensure trailing newline in output

The fold was placing a leading newline, desired to separate the totals
from the logging above. It did not include a trailing newline, which
messes up my prompt when the tool is run.

We could explicitly do that, but it was easier to switch the fold to
place trailing newlines (including a final one) and then have an
explicitly empty first element in our list of lines.

pbrisbin committed:
Clarify close-iteration output

I found the previous output hard to understand:

    Completed
    - new points: 19
    - carried over points: 14
    - new stories: 13
    - carried over stories: 17
    Incomplete
    - points completed: 11
    - carry over points: 22
    - carry over stories: 14

    44 / 66

Some of my confusions:

- What does "new" mean?
- Does "carried over" mean into our out of the closing iteration?
- Ditto "carry over"
- How does "Incomplete" have "completed" points?
- is "44 / 66" velocity? Why make me do the math?

The new output is this:

    Completed points: 33 (14 were carry-over)
    Completed tasks: 30 (17 were carry-over)

    Incomplete points: 22
    Incomplete tasks: 14 (33 total cost)

    Velocity: 44/66 (66.66666666666666%)

This attempts to clarify:

- That we're showing totals and breaking out what was carried
- Past tense when talking about carried into the iteration
- How we get credit for partial points, but show total incomplete too

And computes velocity for you.

pbrisbin commented:
My output doesn't clearly show how the `44` came to be (it's omitting the `11` points of carry we get credit for. Will update.

pbrisbin commented:
Updated:

```
Completed points: 44 (14 carry-over, 19 new, 11 in carrying tasks)
Completed tasks: 30 (17 carry-over, 13 new)

Incomplete points: 22
Incomplete tasks: 14 (worth 33 total)

Velocity: 44/66 (66.66666666666666%)
```

pbrisbin committed:
Fixup output format more

eborden commented:
One change, can you remove the denominator and percentage from this
```
Velocity: 44/66 (66.66666666666666%)
```
We actually don't want to trust those numbers, because we really care about the total commitment at the beginning of the iteration, not the total at the end. That number can change because tasks can be removed and added to the iteration. 

This tool can only effectively give you a "completed points" count.

eborden reviewed (changes requested):
from before :point_up:

pbrisbin commented:
Hmm, happy to change whatever, but I don't really understand what's going on.

If this tool outputs only `44`, where/how do I get velocity?

> we really care about the total commitment at the beginning of the iteration, not the total at the end

Agreed, but we don't allow ourselves to change cost mid-iteration because it screws up this math... right? Should total at start == total at end?

If the denominator can't be trusted why can the numerator?

eborden commented:
Here are the iteration steps:

1. Run `start-iteration`, which gives us the total commitment.
2. Record the total commitment in the iteration tracker.
3. Do the work.
4. Mark carryover.
5. Run `close-iteration`, which gives us the total completed points.
6. Calculate velocity `(completed / commitment) * 100`.

We try to limit removing points mid-iteration, but it sometimes has to happen. We prefer a point for point change if possible. However we do sometimes add points from overflow when work is completed sooner than expected. Velocity is a measure of planning (not completion). So we want to measure our planned commitment against what we actually did. If you don't calculate with `commitment` then you will never observe how changes to the plan inflate or deflate velocity. If someone finished their work half way through and picks up 10 more points, then `completed/commitment` will show a greater than 100% velocity.

pbrisbin commented:
Ok, so the issue for me was Joris handled start of iteration and I handled close. I don't know if he didn't record the commitment, or I just couldn't find it :man_shrugging: .

So I'll change this PR from adding the percentage computation to removing the shown velocity altogether. Sounds like we only want to show completed points, which is its own line.

pbrisbin committed:
Remove output of committed/velocity

These values can only be viewed (and must be recorded) at start of
iteration. Calculating them during close would misrepresent things.

eborden commented:
In that situation you'd run `start-iteration` to produce a commitment (even if it is probably incorrect) and then run `close-iteration` to produce the completed. It is more painful, but the pain is representative of a misstep.
